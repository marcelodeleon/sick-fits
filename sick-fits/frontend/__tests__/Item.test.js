import {shallow} from 'enzyme';
import toJSON from 'enzyme-to-json';
import Item from '../components/Item';

const fakeItem = {
  image: 'somepng.jpg',
  title: 'title',
  id: 'ItemID',
  description: 'Awesome Description',
  price: 123330,
};

describe('<Item/>', () => {
  it('renders', () => {
    shallow(<Item item={fakeItem} />);
  });

  it('matches the snapshot', () => {
    const wrapper = shallow(<Item item={fakeItem} />);

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('updates via props', () => {
    const updatedFakeItem = {...fakeItem, title: 'modified title'};

    const wrapper = shallow(<Item item={fakeItem} />);
    expect(toJSON(wrapper)).toMatchSnapshot();


    const updatedWrapper = shallow(<Item item={updatedFakeItem} />);
    expect(toJSON(updatedWrapper)).toMatchSnapshot();
  })
});
