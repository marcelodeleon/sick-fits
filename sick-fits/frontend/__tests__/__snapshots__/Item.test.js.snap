// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item/> matches the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="title"
    src="somepng.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "ItemID",
          },
        }
      }
    >
      <a>
        title
      </a>
    </Link>
  </Title>
  <PriceTag>
    $1,233.30
  </PriceTag>
  <p>
    Awesome Description
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "ItemID",
          },
        }
      }
    >
      <a>
        Edit
      </a>
    </Link>
    <AddToCart
      id="ItemID"
    />
    <DeleteItem
      id="ItemID"
    />
  </div>
</ItemStyles__Item>
`;

exports[`<Item/> updates via props 1`] = `
<ItemStyles__Item>
  <img
    alt="title"
    src="somepng.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "ItemID",
          },
        }
      }
    >
      <a>
        title
      </a>
    </Link>
  </Title>
  <PriceTag>
    $1,233.30
  </PriceTag>
  <p>
    Awesome Description
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "ItemID",
          },
        }
      }
    >
      <a>
        Edit
      </a>
    </Link>
    <AddToCart
      id="ItemID"
    />
    <DeleteItem
      id="ItemID"
    />
  </div>
</ItemStyles__Item>
`;

exports[`<Item/> updates via props 2`] = `
<ItemStyles__Item>
  <img
    alt="modified title"
    src="somepng.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "ItemID",
          },
        }
      }
    >
      <a>
        modified title
      </a>
    </Link>
  </Title>
  <PriceTag>
    $1,233.30
  </PriceTag>
  <p>
    Awesome Description
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "ItemID",
          },
        }
      }
    >
      <a>
        Edit
      </a>
    </Link>
    <AddToCart
      id="ItemID"
    />
    <DeleteItem
      id="ItemID"
    />
  </div>
</ItemStyles__Item>
`;
